[Core] Rich logging for not so rich people

Kowlin opened the pull request:
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes

Added rich logging and a flag to forcefully enable it within red. Rich logging will beautify your terminal so it all looks shiny and new.

![image](https://user-images.githubusercontent.com/10947836/98176284-82327680-1ef8-11eb-8915-47f17e395740.png)

![image](https://user-images.githubusercontent.com/10947836/98176335-9b3b2780-1ef8-11eb-9a87-a3120fc19500.png)

closes #3936

Kowlin committed:
Pop stash

Kowlin committed:
add rich to setup

Kowlin committed:
Added forceful enabling of rich logging

Kowlin committed:
revert some unintended pushed

Kowlin committed:
Fix possible unbound var
Fix possible 0 members w/out members intent

Kowlin committed:
One day I won't forget to do style passes

Kowlin committed:
So this is a thing apperently...

madebylydia commented:
Guess `Loaded X cogs with Y commands` could be put under `Unique Users` as it's a part of the bot's statistics, even if not directly related to Discord, I think it would fit better to me.
Maybe putting the Invite URL on the middle could be nice too. Feel free to correct me.

Kowlin committed:
Bump rich to 9.5.1

Jackenmen committed:
Lock secondary deps

Jackenmen committed:
Merge branch 'V3/develop' into pr/feature/rich-formatting

Jackenmen committed:
Different stuff, see the full commit description for more info

- Change few things from print to log.info
- put the log handlers on the root logger instead of individual loggers
- capture warnings to a logger

Kowlin committed:
Modify log handler to show logger name

Kowlin committed:
Add a Triboolian to force disable rich

Kowlin committed:
Style checks

Kowlin committed:
shortened time, added logger name... again.

Jackenmen committed:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

Kowlin committed:
Style & linking

Kowlin committed:
Merge branch 'pr/feature/rich-formatting' of github.com:Kowlin/Red-DiscordBot into pr/feature/rich-formatting

Jackenmen committed:
Be or not to be? Whatever man, it's 4:30 in the morning, goto sleep >.<

Kowlin committed:
Reintroduce outdated message.

Kowlin committed:
Merge branch 'pr/feature/rich-formatting' of github.com:Kowlin/Red-DiscordBot into pr/feature/rich-formatting

Jackenmen reviewed (approved):
If this breaks, you can blame *Rich* people for not sponsoring us :)

https://patreon.com/Red_Devs
